src/main/java/io/swagslash/gametrackerserver/service/impl/AuthenticationServiceImpl.java

```diff
@@ -59,6 +70,27 @@ public AuthResponse loginUser(LoginRequest request) {
         return new AuthResponse(token, user.getUsername(), appProperties.getAuth().getTokenExpirationMsec(), user.getEmail());
     }
 
+    @Override
+    public boolean isValidAgentKey(String key) throws BadRequestException {
+        AgentToken token = tokenRepository.findByToken(key);
+        if(token == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public String getAgentToken() throws BadRequestException {
```

Comment:
Should be `createAgentToken` as you create a new token and do not query it from the `tokenRepository`